docs/paper/dev/api/how-do-plugins-work.md

```diff
@@ -0,0 +1,101 @@
+---
+slug: /dev/how-do-plugins-work
+description: How plugins work in Paper.
+---
+
+# How Plugins Work
+
+## Introduction
+
+Plugins are a way to extend the functionality of a Minecraft server. They are written in JVM-based languages such as 
+Java, Kotlin, Groovy or Scala. Plugins are loaded from the `plugins` folder in the server directory. Plugins will be 
+loaded from a `.jar` file. 
+
+## Plugin Lifecycle
+
+Plugins are loaded and unloaded at runtime. When a plugin is loaded, it is initialized and enabled. When a plugin is 
+unloaded, it is disabled and finalized.
+
+### Initialization
+
+When a plugin is loaded, it is initialized. This means that the plugin is loaded into memory and its `onLoad` 
+method is called. This method is used to initialize the plugin and set up any resources that it needs. Most of the
+Bukkit API is not available at this point, so it is not safe to interact with it.
+
+### Enabling
+
+When a plugin is enabled, its `onEnable` method is called. This method is used to set up any resources that the plugin 
+needs to run. This method is called when the plugin is initialized but before the server has started ticking, so it is 
+safe to register event listeners and other resources that the plugin needs to run, however often not safe to interact 
+with a lot of APIs.
+
+This is when you can also open database connections, start threads, and other things that are not safe to do in the
+`onLoad` method.
+
+### Disabling
+
+When a plugin is disabled, its `onDisable` method is called. This method is used to clean up any resources that the 
+plugin has allocated. This method is called before all plugins are unloaded, and is meant for any cleanup that needs to
+be done before the plugin is unloaded. This may include saving data to disk or closing connections to databases.
+
+## Event Listeners
+
+Events are a way for plugins to listen to things that happen in the server and run code when they are fired. For 
+example, the `PlayerJoinEvent` is fired when a player joins the server. This is a more performant way to run code when 
+something happens, as opposed to constantly checking.
+
+Some events are cancellable. This means that when the event is fired, it can be cancelled which negates or stops the 
+effect of the event. For example, the `PlayerMoveEvent` is cancellable. This means that when it is cancelled, the player 
+will not move. This is useful for things like anti-cheat, where you want to cancel the event if the player is moving too fast.
+
+It is important to think about how "hot" an event is when writing event listeners. A "hot" event is an event that is fired
+very often. For example, the `PlayerMoveEvent` is fired every time a player moves. This means that if you have a lot of
+expensive code in your event listener, it will be run every time a player moves. This can cause a lot of lag. It is
+important to keep event listeners as lightweight as possible.
```

Comment:
I would add something about doing some checks first. i.e. in player move event, you often only care if they actually moved to a new block, so check that first and early return